salt/client/__init__.py

```diff
@@ -806,48 +808,16 @@ def get_returns_no_block(
 
         Yield either the raw event data or None
 
-       Pass a list of additional regular expressions as `tags_regex` to search
-       the event bus for non-return data, such as minion lists returned from
-       syndics.
+        Pass a list of additional regular expressions as `tags_regex` to search
+        the event bus for non-return data, such as minion lists returned from
+        syndics.
         '''
-        if event is None:
-            event = self.event
-
-        jid_tag = 'salt/job/{0}'.format(jid)
-        jid_tag_regex = '^salt/job/{0}'.format(jid)
-
-        tag_search = []
-        tag_search.append(re.compile(jid_tag_regex))
-        if isinstance(tags_regex, str):
-            tag_search.append(re.compile(tags_regex))
-        elif isinstance(tags_regex, list):
-            for tag in tags_regex:
-                tag_search.append(re.compile(tag))
+
         while True:
-            if self.opts.get('transport') == 'zeromq':
-                try:
-                    raw = event.get_event_noblock()
-                    if gather_errors:
-                        if (raw and
-                                (raw.get('tag', '').startswith('_salt_error') or
-                                any([tag.search(raw.get('tag', '')) for tag in tag_search]))):
-                            yield raw
-                    else:
-                        if raw and raw.get('tag', '').startswith(jid_tag):
-                            yield raw
-                        else:
-                            yield None
-                except zmq.ZMQError as ex:
-                    if ex.errno == errno.EAGAIN or ex.errno == errno.EINTR:
-                        yield None
-                    else:
-                        raise
-            else:
-                raw = event.get_event_noblock()
-                if raw and raw.get('tag', '').startswith(jid_tag):
-                    yield raw
-                else:
-                    yield None
+            # TODO(driskell): This was previously completely nonblocking.
+            #                 Should get_event have a nonblock option?
```

Comment:
Hehe :)

`get_event_noblock` is the counterpart of `get_event_block`

`get_event` is higher level, has extra options and parameters, and sites on top of `_get_event` which then sites on top of `get_event_block`: https://github.com/saltstack/salt/blob/2015.8/salt/utils/event.py#L406

So to get the same functionality but completely nonblocking there would be a need to have an option to `get_event` that is passed down the layers to `_get_event` so it can call `get_event_noblock` instead of `get_event_block`.

I didn't want to change more than I had to though :+1: :-)